runtime/drivers/duckdb/connectors.go

```diff
@@ -51,11 +94,23 @@ func (c *connection) ingestFiles(ctx context.Context, source *connectors.Source,
 		format = value.(string)
 	}
 
-	from, err := sourceReader(filenames, delimiter, format)
+	hivePartition := 1
+	if value, ok := source.Properties["hive_partitioning"]; ok {
+		if !value.(bool) {
+			hivePartition = 0
+		}
+	}
+
+	from, err := sourceReader(filenames, delimiter, format, hivePartition)
 	if err != nil {
 		return err
 	}
+
 	query := fmt.Sprintf("CREATE OR REPLACE TABLE %s AS (SELECT * FROM %s);", source.Name, from)
+	if appendToTable {
+		query = fmt.Sprintf("INSERT INTO %s (SELECT * FROM %s);", source.Name, from)
+	}
```

Comment:
1. Agreed. 
2. Did not think about it. Thanks for pointing out. I do remember reading about bad performance of bulk inserts in duck db as well. Will try to test performance and will change if required.

Reply:
I ran explain analyse on copy vs insert and both gave similar numbers on a insert of ~6GB parquet file. Seems like the bad performance of bulk inserts comes into picture when we are loading rows in our application and passing it to duckdb.
Attaching the results of explain analyze statements.
[results.txt](https://github.com/rilldata/rill-developer/files/10533840/results.txt)

COPY also has following limitations wrt insert:
1. It doesn't seem to support copying from multiple files(unless we can come with some regex to represent those files). I could not neither find an example of copying from multiple files nor could figure out the syntax. 
2. We need to  write duplicated logic of generating query with all the parameters like delimiter, hive_parition etc.

I think we can go ahead with the insert statement itself. Please let me know your thoughts.

Reply:
Excellent, so the poor performance you previously observed was due to inserting rows from application instead of loading files?

The results look pretty clear that INSERT has same performance, so since it has better syntax/options, let's stick with that

Reply:
No. So the performance degradation was related to loading each file one by one and not passing all files in one go. 

Yeah will keep the INSERT.